subsys/net/openthread/rpc/client/ot_rpc_netdiag.c

```diff
@@ -6,13 +6,288 @@
+		aNetworkDiagTlv->mData.mChildTable.mCount = nrf_rpc_decode_uint(ctx);
+		zcbor_list_start_decode(ctx->zs);
+		for (int i = 0; i < aNetworkDiagTlv->mData.mChildTable.mCount; i++) {
+			uint8_t mode;
+
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mTimeout =
+				nrf_rpc_decode_uint(ctx);
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mLinkQuality =
+				nrf_rpc_decode_uint(ctx);
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mChildId =
+				nrf_rpc_decode_uint(ctx);
+			mode = nrf_rpc_decode_uint(ctx);
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mMode.mRxOnWhenIdle =
+				(bool)(mode & BIT(0));
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mMode.mDeviceType =
+				(bool)(mode & BIT(1));
+			aNetworkDiagTlv->mData.mChildTable.mTable[i].mMode.mNetworkData =
+				(bool)(mode & BIT(2));
+		}
+		zcbor_list_end_decode(ctx->zs);
+		break;
+	default:
+		break;
+	}
+}
+
+otError otThreadGetNextDiagnosticTlv(const otMessage *aMessage, otNetworkDiagIterator *aIterator,
+				     otNetworkDiagTlv *aNetworkDiagTlv)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	otError error;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, sizeof(ot_msg_key) + 2 + sizeof(otNetworkDiagIterator));
+	nrf_rpc_encode_uint(&ctx, (ot_msg_key)aMessage);
+	nrf_rpc_encode_uint(&ctx, *aIterator);
+	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, OT_RPC_CMD_THREAD_GET_NEXT_DIAGNOSTIC_TLV, &ctx);
+
+	error = nrf_rpc_decode_uint(&ctx);
+
+	if (error != OT_ERROR_NONE) {
+		goto exit;
+	}
+
+	*aIterator = nrf_rpc_decode_uint(&ctx);
+	ot_rpc_decode_network_diag_tlv(&ctx, aNetworkDiagTlv);
+
+exit:
+	if (!nrf_rpc_decoding_done_and_check(&ot_group, &ctx)) {
+		ot_rpc_report_rsp_decoding_error(OT_RPC_CMD_THREAD_GET_NEXT_DIAGNOSTIC_TLV);
+	}
+
+	return error;
+}
+
+static void ot_rpc_cmd_thread_send_diagnostic_get_cb(const struct nrf_rpc_group *group,
+						     struct nrf_rpc_cbor_ctx *ctx,
+						     void *handler_data)
+{
+	otError error;
+	otMessage *message;
+	otMessageInfo message_info;
+
+	error = nrf_rpc_decode_uint(ctx);
+	message = (otMessage *)nrf_rpc_decode_uint(ctx);
+	ot_rpc_decode_message_info(ctx, &message_info);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_cmd_decoding_error(OT_RPC_CMD_THREAD_SEND_DIAGNOSTIC_GET_CB);
+		return;
+	}
+
+	if (receive_diag_get_cb != NULL) {
+		receive_diag_get_cb(error, message, &message_info, receive_diag_get_cb_context);
+	}
+
+	nrf_rpc_rsp_send_void(group);
+}
+
+otError otThreadSendDiagnosticGet(otInstance *aInstance, const otIp6Address *aDestination,
+				  const uint8_t *aTlvTypes, uint8_t aCount,
+				  otReceiveDiagnosticGetCallback aCallback, void *aCallbackContext)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	otError error;
+
+	ARG_UNUSED(aInstance);
+
+	receive_diag_get_cb = aCallback;
+	receive_diag_get_cb_context = aCallbackContext;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 1 + OT_IP6_ADDRESS_SIZE + 4 + aCount * 2);
+	nrf_rpc_encode_buffer(&ctx, (const char *)aDestination, OT_IP6_ADDRESS_SIZE);
```

Comment:
`nrf_rpc_encode_buffer` takes `const void*`. Is cast to `const char*` needed?

Reply:
yeah, I think I copied this line from somewhere and forgot to align it. changed to passing `aDestination->mFields.m8` as well as IMO it looks more readable